### DISCUSSION TITLE
rector using PHPStan prefixed nette/utils

### DISCUSSION THREAD

**Discussion by staabm** [rectorphp/rector#7981] [General]
I am wondering why rector is using PHPStan prefixed nette/utils, even though it declares a dependency on nette/utils itself

<img width="1064" alt="grafik" src="https://github.com/rectorphp/rector/assets/120441/c2deb1b7-7277-4752-b2aa-674707e9a671">

background of my question is, that [I want to test some nette/utils improvements](https://github.com/nette/utils/pull/296), but this is kind of hard because of rector is using the PHPStan prefixed classes

**Comment by TomasVotruba** [comment]
Hey, thanks for the improvement!

How many of nette/utils classes we use? If just a couple, I think it might be worth it to port them directly to Rector, so we can make them simple and fast :)

**Reply by staabm** [reply]
I think we mostly use them for DI indirect via PHPStan.

I saw your experiments with laravel DI: in case these will get more broad application, we still need nette DI for PHPStan, right?

**Reply by TomasVotruba** [reply]
> I think we mostly use them for DI indirect via PHPStan.

I see :+1: 

> I saw your experiments with laravel DI: in case these will get more broad application, we still need nette DI for PHPStan, right?

Yes, you're right. The part I want to check if laravel/di would be faster/cheaper for Rector. I feel like Symfony is great for web application, but not for light CLI like we have here.

**Reply by staabm** [reply]
We need to think about, whether its worth it to have a diffferent DI then phpstan itself is using.

from a runtime performance point of view, the current solution is slow because of configurable rules.
in recent PHPStan releases the DI nearly does not show up in profiles. Maybe we are using it not as designed.

**Reply by TomasVotruba** [reply]
You've good points. Having real number for using nette/di and laravel/di would be very helpful here :+1:

**Comment by staabm** [comment]
seems the change is not worth it
